File: app/src/main/java/se/curtrune/lucy/classes/Reward.java
package se.curtrune.lucy.classes;

import com.google.gson.Gson;

import java.io.Serializable;

public class Reward implements Serializable {
    private String message;
    public enum Type{
        AFFIRMATION, USER_DEFINED, CONFETTI
    }
    private Type type;

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }
    public String toJson(){
        return new Gson().toJson(this);
    }
}

File: app/src/main/java/se/curtrune/lucy/classes/Reward.kt
package se.curtrune.lucy.classes

import com.google.gson.Gson
import java.io.Serializable

@kotlinx.serialization.Serializable
class Reward {
    private val message: String? = null

    enum class Type {
        AFFIRMATION, USER_DEFINED, CONFETTI
    }

    var type: Type? = null

    fun toJson(): String? {
        return Gson().toJson(this)
    }
}
